app/questionnaire/questionnaire_store_updater.py

```diff
@@ -321,6 +322,52 @@ def _capture_section_dependencies_for_answer(self, answer_id: str) -> None:
             else:
                 self.dependent_sections.add(DependentSection(section_id, None, None))
 
+    def _capture_section_dependencies_progress_value_source_for_current_section(
+        self,
+    ) -> None:
+        """
+        Captures a unique list of section ids that are dependents of the current section, for progress value sources.
+        """
+        dependent_sections = [
+            section_key
+            for section_key, dependent_sections in self._schema.when_rules_section_dependencies_by_section_for_progress_value_source.items()
+            if any(
+                self._current_location.section_id in section_ids
+                for section_ids in dependent_sections
+            )
+        ]
```

Comment:
This applies to the method below also. Why are we looping through every section dependency?. It should be a look-up based on section id for this and block id for the method below. So for a given section/block id, it should return the list of dependent sections.

Reply:
I've updated these methods again now 👍